--- a/mscore/CMakeLists.txt	2009-10-18 21:55:54.780493407 +0200
+++ b/mscore/CMakeLists.txt	2009-10-18 22:05:20.954469144 +0200
@@ -153,13 +153,22 @@
 if (APPLE OR MINGW)
       set (USE_JACK 0)
 else (APPLE OR MINGW)
-      PKGCONFIG (jack 0.98.0 JACK_INCDIR JACK_LIBDIR JACK_LIB JACK_CPP )
-endif (APPLE OR MINGW)
-
-if (JACK_INCDIR)
-	message("Jack detected")
-      set (USE_JACK 1)
-endif (JACK_INCDIR)
+	SET(BUILD_JACK AUTO CACHE STRING "Build with support for jack (jack audio connection kit). jack >= 0.98.0 needed. [ON/OFF/AUTO]")
+	IF(BUILD_JACK)
+		PKGCONFIG(jack 0.98.0 JACK_INCDIR JACK_LIBDIR JACK_LIB JACK_CPP)
+		IF(JACK_INCDIR)
+			MESSAGE(STATUS "jack (jack audio connection kit) >= 0.98.0 found")
+			SET(USE_JACK 1)
+		ELSE()
+			MESSAGE(STATUS "jack (jack audio connection kit) >= 0.98.0 not found")
+			IF(NOT BUILD_JACK STREQUAL AUTO)
+				MESSAGE(SEND_ERROR "Error: jack requested but not found (BUILD_JACK=${BUILD_JACK})")
+			ENDIF()
+		ENDIF()
+	ELSE()
+		MESSAGE(STATUS "jack (jack audioconnection kit) disabled")
+	ENDIF()
+endif ()
 
 ##
 ## portaudio
